File: pwiz_tools/Skyline/Executables/Tools/ToolServiceCmd/ExampleTool/CountProteins.R
report_proteinlocators = '<?xml version="1.0"?>
<views>
  <view name="ProteinNames" rowsource="pwiz.Skyline.Model.Databinding.Entities.Protein" sublist="Results!*">
    <column name="Name" />
  </view>
</views>'

tool_folder <- commandArgs(trailingOnly = TRUE)[1];
cat ("Tool folder ", tool_folder);
connectionname <- commandArgs(trailingOnly = TRUE)[2];
program_path <- file.path(tool_folder, "ToolServiceCmd.exe")
csv_output <- system2(program_path, c("GetReport", "--connectionname", connectionname), input=report_proteinlocators, stdout=TRUE);
df <- read.csv(text=csv_output, stringsAsFactors = FALSE);

cat("There are ", nrow(df), " proteins");

File: pwiz_tools/Skyline/Executables/Tools/ToolServiceCmd/ExampleTool/CountProteins.py
#!/usr/bin/python

import os
import sys
import subprocess
report_proteinlocators = """<?xml version="1.0"?>
<views>
  <view name="ProteinLocators" rowsource="pwiz.Skyline.Model.Databinding.Entities.Protein" sublist="Results!*">
    <column name="Name" />
  </view>
</views>"""

connectionname = sys.argv[1]
toolservicecmdexe = os.path.dirname(os.path.realpath(__file__)) + "/bin/ToolServiceCmd.exe"

args=[toolservicecmdexe, 'GetReport', '--connectionname', connectionname]
print "Executing command ", args
p = subprocess.Popen([toolservicecmdexe, 'GetReport', '--connectionname', connectionname], stdin=subprocess.PIPE, stdout=subprocess.PIPE)
(report_text, error_text) = p.communicate(input=report_proteinlocators)
print "Result of command was", len(report_text), "characters long."

print "There are", report_text.count('\n') - 1, "proteins"
